synapse/http/server.py

```diff
@@ -298,7 +298,7 @@ def _async_render(self, request):
         # installed by @request_handler.
 
         kwargs = intern_dict({
-            name: urllib.unquote(value).decode("UTF-8") if value else value
```

Comment:
this looks wrong: we deliberately utf-8 decode query params here. "%F0%9F%92%A9" should give you a smiley poo, which python 2's urllib.unquote doesn't do.

In fact, I'm surprised we don't have a unit test which picks this up, since I'm fairly sure we've been bitten by it before. Any chance you could add something, possibly by factoring bits of this out?